core/drivers/ivshmem.c

```diff
@@ -559,47 +631,47 @@ static void qnx_ivshmem_init(void)
 	g_ivshmem_devs[0].smc_addr = (vaddr_t)phys_to_virt(shmem_addr, MEM_AREA_RAM_NSEC);
 
 	smc_evt_src = (uint32_t *)g_ivshmem_devs[0].smc_addr;
-	smc_vm_ids = (struct optee_vm_ids *)(g_ivshmem_devs[0].smc_addr +
+	smc_vm_ids[0] = (struct optee_vm_ids *)(g_ivshmem_devs[0].smc_addr +
```

Comment:
is it the same logic but the g_ivshmem_dev_num = 1? if yes, use same function with g_ivshmem_dev_num = 1